src/antq/util/git.clj

```diff
@@ -1,19 +1,39 @@
 (ns antq.util.git
   (:require
+   [antq.log :as log]
    [clojure.java.shell :as sh]
    [clojure.string :as str]))
 
 (defn- extract-tags
   [ls-remote-resp]
-  (->> (:out ls-remote-resp)
+  (some->> (:out ls-remote-resp)
        (str/split-lines)
        (keep #(second (str/split % #"\t" 2)))
        (filter #(= 0 (.indexOf ^String % "refs/tags")))
        (map #(str/replace % #"^refs/tags/" ""))))
 
+(defn- sh-git-ls-remote
+  [url]
+  (loop []
+    (let [{:keys [exit err] :as res} (sh/sh "git" "ls-remote" url)]
+      (cond
+        (= 0 exit)
+        res
+
+        (and (< 0 exit) (not (str/includes? err "Operation timed out")))
+        (log/error (str "git ls-remote failed on: " url))
+
+        :else
+        (do
+          (log/error "git ls-remote timed out, retrying")
+          (recur))))))
```

Comment:
I'd like to avoid infinite loops, so could you set an upper limit on the number of retries?
A fixed number is fine.